server/assertions/operation.go

```diff
@@ -0,0 +1,156 @@
+package assertions
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/kubeshop/tracetest/server/model"
+	"github.com/kubeshop/tracetest/server/traces"
+)
+
+type ExpressionOperation interface {
+	Execute(model.LiteralValue, model.LiteralValue) (model.LiteralValue, error)
+}
+
+type sumOperation struct{}
+
+func (*sumOperation) Execute(value1 model.LiteralValue, value2 model.LiteralValue) (model.LiteralValue, error) {
+	return runMathOperationOnNumbers(value1, value2, func(f1, f2 float64) float64 {
+		return f1 + f2
+	})
+}
+
+var _ ExpressionOperation = &sumOperation{}
+
+type subtractOperation struct{}
+
+func (*subtractOperation) Execute(value1 model.LiteralValue, value2 model.LiteralValue) (model.LiteralValue, error) {
+	return runMathOperationOnNumbers(value1, value2, func(f1, f2 float64) float64 {
+		return f1 - f2
+	})
+}
+
+var _ ExpressionOperation = &subtractOperation{}
+
+type multiplyOperation struct{}
+
+func (*multiplyOperation) Execute(value1 model.LiteralValue, value2 model.LiteralValue) (model.LiteralValue, error) {
+	return runMathOperationOnNumbers(value1, value2, func(f1, f2 float64) float64 {
+		return f1 * f2
+	})
+}
+
+var _ ExpressionOperation = &multiplyOperation{}
+
+type divideOperation struct{}
+
+func (*divideOperation) Execute(value1 model.LiteralValue, value2 model.LiteralValue) (model.LiteralValue, error) {
+	return runMathOperationOnNumbers(value1, value2, func(f1, f2 float64) float64 {
+		return f1 / f2
+	})
+}
+
+var _ ExpressionOperation = &divideOperation{}
+
+func runMathOperationOnNumbers(value1 model.LiteralValue, value2 model.LiteralValue, operation func(float64, float64) float64) (model.LiteralValue, error) {
+	if err := validateFieldType(value1); err != nil {
+		return model.LiteralValue{}, err
+	}
+
+	if err := validateFieldType(value2); err != nil {
+		return model.LiteralValue{}, err
+	}
+
+	operationType := "number"
+	if value1.Type == "duration" {
+		value1.Value = fmt.Sprintf("%d", traces.ConvertTimeFieldIntoNanoSeconds(value1.Value))
+		operationType = "duration"
+	}
+
+	if value2.Type == "duration" {
+		value2.Value = fmt.Sprintf("%d", traces.ConvertTimeFieldIntoNanoSeconds(value2.Value))
+		operationType = "duration"
+	}
+
+	number1, _ := strconv.ParseFloat(value1.Value, 64)
+	number2, _ := strconv.ParseFloat(value2.Value, 64)
+
+	result := operation(number1, number2)
+	resultStr := fmt.Sprintf("%d", int64(result))
+	if strings.Contains(value1.Value, ".") || strings.Contains(value2.Value, ".") {
+		// float number
+		resultStr = fmt.Sprintf("%.2f", result)
+	}
+
+	return model.LiteralValue{
+		Value: resultStr,
+		Type:  operationType,
+	}, nil
+}
+
+func validateFieldType(field model.LiteralValue) error {
+	if field.Type != "number" && field.Type != "duration" {
+		return fmt.Errorf("operation is only allowed on numbers and duration fields")
+	}
+
+	return nil
+}
+
+func getOperationExecutor(operation string) (ExpressionOperation, error) {
+	switch operation {
```

Comment:
This switch looks weird. I would either return a basic func that does the operation from here, or have a "registry" approach for extensibility. Otherwise, there's a lot of code overhead to end up with a small set of not-changing options